data_acquisition/py/gee_functions.py

```diff
@@ -578,14 +991,40 @@ def ref_pull_89_DSWE3(image):
   f = cf_mask(image).select('cfmask')
   # process image with st SR cloud mask
   a = sr_aerosol(image).select('medHighAero')
-  # where the f mask is > 2 (clouds and cloud shadow), call that 1 (otherwise 0) and rename as clouds.
+  # where the f mask is > 1 (clouds and cloud shadow), call that 1 (otherwise 0) and rename as clouds.
   clouds = f.gte(1).rename('clouds')
   #apply dswe function
   d = DSWE(image).select('dswe')
-  pCount = d.gt(0).rename('dswe_gt0').updateMask(f.eq(0)).updateMask(r.eq(1)).selfMask()
-  dswe1 = d.eq(1).rename('dswe1').updateMask(f.eq(0)).updateMask(r.eq(1)).selfMask()
+  gt0 = (d.gt(0).rename('dswe_gt0')
+    .updateMask(f.eq(0))
+    .updateMask(r.eq(1))
+    .selfMask()
+    )
+  dswe1 = (d.eq(1).rename('dswe1')
+    .updateMask(f.eq(0))
+    .updateMask(r.eq(1))
+    .selfMask()
+    )
   # band where dswe is 3 and apply all masks
-  dswe3 = d.eq(3).rename('dswe3').updateMask(f.eq(0)).updateMask(r.eq(1)).selfMask()
+  dswe3 = (d.eq(3).rename('dswe3')
+    .updateMask(f.eq(0))
+    .updateMask(r.eq(1))
+    .selfMask()
+    )
+  # define dswe 1a where d is not 0 and red/green threshold met
+  grn_alg_thrsh = image.select('Green').gt(0.05)
+  red_alg_thrsh = image.select('Red').lt(0.04)
+  alg = (d.gt(1).rename('algae')
+    .And(grn_alg_thrsh.eq(1))
+    .And(red_alg_thrsh.eq(1))
+    )
+  dswe1a = (d.eq(1)
+    .Or(alg.eq(1))
+    .rename('dswe1a')
+    .updateMask(f.eq(0))
+    .updateMask(r.eq(1))
+    .selfMask()
+    )
```

Comment:
question: is there a reason why f.eq(0) and r.eq(0) are set but the hs.eq(1) is left out and called later on?

Reply:
Oversight! Adding in now.